app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java

```diff
@@ -123,48 +180,57 @@ private void setupFocusedFavorListen() {
               try {
-                if (favor != null) {
+                if (favor != null && favorViewModel.isShowObservedFavor()) {
+                  favorViewModel.setShowObservedFavor(false);
                   focusedFavor = favor;
                   boolean isRequested = // check if favor is requested
                       favor
                           .getRequesterId()
                           .equals(DependencyFactory.getCurrentFirebaseUser().getUid());
-
-                  Marker marker = drawFavorMarker(focusedFavor, isRequested);
+                  boolean isEdited = focusedFavor.getStatusId() == FavorStatus.EDIT.toInt();
+                  Marker marker = drawFavorMarker(focusedFavor, isRequested, isEdited);
+                  if (isEdited) newMarkers.add(marker);
                   marker.showInfoWindow();
-                  focusViewOnLocation(focusedFavor.getLocation());
+                  focusViewOnLocation(focusedFavor.getLocation(), true);
                 }
               } catch (Exception e) {
                 CommonTools.showSnackbar(requireView(), getString(R.string.error_database_sync));
               }
             });
   }
 
-  private Marker drawFavorMarker(Favor favor, boolean isRequested) {
+  private Marker drawFavorMarker(Favor favor, boolean isRequested, boolean isEdited) {
     LatLng latLng =
         new LatLng(favor.getLocation().getLatitude(), favor.getLocation().getLongitude());
     float markerColor =
         isRequested ? BitmapDescriptorFactory.HUE_AZURE : BitmapDescriptorFactory.HUE_RED;
+    String markerTitle =
+        (isEdited && favor.getTitle().equals(""))
+            ? "Drag marker to choose position"
+            : favor.getTitle();
+    String markerDescription = isEdited ? "Click window to request favor" : favor.getDescription();
     Marker marker =
         mMap.addMarker(
             new MarkerOptions()
                 .position(latLng)
-                .title(favor.getTitle())
-                .icon(BitmapDescriptorFactory.defaultMarker(markerColor)));
-    marker.setSnippet(favor.getDescription());
+                .title(markerTitle)
+                .icon(BitmapDescriptorFactory.defaultMarker(markerColor))
+                .snippet(markerDescription));
     marker.setTag(
         new ArrayList<Object>() {
           {
             add(favor.getId());
             add(isRequested);
           }
         });
+    if (isEdited) marker.setDraggable(true);
     return marker;
   }
 
-  private void focusViewOnLocation(Location location) {
+  private void focusViewOnLocation(Location location, boolean animate) {
     LatLng latLng = new LatLng(location.getLatitude(), location.getLongitude());
-    mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, mMap.getMaxZoomLevel() - 5));
+    if (animate) mMap.animateCamera(CameraUpdateFactory.newLatLng(latLng));
+    else mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, mMap.getMaxZoomLevel() - 5));
```

Comment:
For next sprint we should customize the zoom size to be equal to the preferred user distance (just thought about it so leaving the comment so we don't forget

Reply:
good idea